src/win.rs

```diff
@@ -103,6 +104,19 @@ impl IntoRawHandle for ::Handle {
     }
 }
 
+impl Hash for Handle {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match self.key {
+            Some(ref k)  =>
+                {   k.volume.hash(state);
+                    k.idx_high.hash(state);
+                    k.idx_low.hash(state);
+                },
+            None => (),
+        }
```

Comment:
I think you can probably `derive(Hash)` on the `Key` type, no? And then this whole body just becomes `self.key.hash(state);`.